packages/labbook/src/components/runtime.tsx

```diff
@@ -193,7 +198,7 @@ export const Runtime: React.FC<IProps> = ({ authoredState, interactiveState, set
       const item = pEntries?.find(i => i.id === id);
       if (item) {
         const updatedEntry: ILabbookEntry = deepmerge(item, fields) as ILabbookEntry;
-        const newEntries = pEntries.map(i => i.id === selectedId ? updatedEntry : i);
+        const newEntries = pEntries.map(i => i === item ? updatedEntry : i);
```

Comment:
Hi Piotr, this was necessary because looking at the selectedId was causing issues when we were creating a new item. When the upload is in progress in this case, we add a new item, which updates the selectedItem, but this code is being hit before that update happens, and the older selected item was being updated as well as the new one.

Reply:
I see.

> As for the second point about checking id's, I though because of how item is defined on line 198, we could just use the item that way. But let me know if you think I should change it.

That's true, it cannot fail currently because of the line 198 and the fact that nothing happens to item between. 👍 

However, if we have access to any kind of `id`, it can be usually safer to use that. Just like a rule of thumb. Imagine someone changes line 198 to:
`const item = { ...pEntries?.find(i => i.id === id), oldItem: true };`
Ofc it doesn't make sense here, but just an example. That's enough to break comparisons relying on reference.